organizing_hub/signals.py

```diff
@@ -128,17 +128,19 @@ def event_promotion_post_save_handler(instance, **kwargs):
     event_promotion = instance
     status = event_promotion.status
     contact_list = event_promotion.contact_list
-    if status == EventPromotionStatus.approved.value[0] and contact_list is None:
-
-        """Create new contact list and add to event promotion"""
-        list_name = 'List for Event Promotion: ' + str(event_promotion)
-        contact_list = ContactList.objects.create(name=list_name)
-        event_promotion.contact_list = contact_list
-        event_promotion.save()
-
-        """Call async task to build list"""
-        build_contact_list_for_event_promotion.delay(event_promotion.id)
-
+    if status == EventPromotionStatus.approved.value[0]:
+        if contact_list is None:
+            """Create new contact list and add to event promotion"""
+            list_name = 'List for Event Promotion: ' + str(event_promotion)
+            contact_list = ContactList.objects.create(name=list_name)
+            event_promotion.contact_list = contact_list
+            event_promotion.save()
+
+            """Call async task to build list"""
+            build_contact_list_for_event_promotion.delay(event_promotion.id)
+        else:
+            """Resave contact list to trigger post_save logic"""
+            contact_list.save()
```

Comment:
Why did you add this `else` clause? I don't understand the purpose. Why do we need to save the contact list if promo is approved and contact list is not `None`?

Reply:
I'm worried this may have side effects. As written previously, it would only do something if promo was approved with no contact list. Now it does stuff if that is the case, or if contact list is already there. But if contact list is not `None`, then it should already have started to build list and then send emails, so why are we triggering anything additional in this case?

Reply:
When an event promo is marked as event not approved, an admin might go and approve the event, and then come back and mark the event promo as approved. This should trigger send again. As-is, the first time an event is marked as approved a list is built, but the send method will not send based on many conditions (such as if the event isn't approved). Some of these conditions aren't permanent, however, like event not approved. Perhaps instead of following precedent and returning from send event function if conditions don't pass we should instead not build list until promo is ready to go out.

Reply:
@cjmabry the prior version of logic was meant to avoid accidentally resending to same list twice, or sending to a stale list. For admin retry send, the process outlined in https://ourrevolution.atlassian.net/browse/TECH-1344 is to set contact list to None and re-approve the promotion. This ensures that a fresh list is regenerated.

Reply:
The other issue with the change you suggested is that it's confusing. It saves contact list for no clear reason. So even with that route, would be better to add more code comments, or call the method needed directly rather than indirectly via a post save trigger.